src/nodes.coffee

```diff
@@ -3498,6 +3506,35 @@ exports.Op = class Op extends Base
   toString: (idt) ->
     super idt, @constructor.name + ' ' + @operator
 
+  ast: ->
+    @checkUpdateAssignability()
+    super()
+
+  astType: ->
+    switch @operator
+      when 'new'           then 'NewExpression'
+      when '||', '&&', '?' then 'LogicalExpression'
+      when '++', '--'      then 'UpdateExpression'
+      else
+        if @isUnary()      then 'UnaryExpression'
+        else                    'BinaryExpression'
+
+  astProperties: ->
```

Comment:
Here, `astProperties()` was a definite improvement on the version using `astChildren()` and `astProps()`, which had parallel conditional structures (so if eg you wanted to see what  a `new` AST looked like, you'd have to scan both `astChildren()` and `astProps()` and mentally combine the results)